packages/backend/src/server/api/endpoints/ap/show.ts

```diff
@@ -102,23 +117,64 @@ export const paramDef = {
 	type: 'object',
 	properties: {
 		uri: { type: 'string' },
+		onlyUriFetch: { type: 'boolean' },
 	},
 	required: ['uri'],
 } as const;
 
 @Injectable()
 export default class extends Endpoint<typeof meta, typeof paramDef> { // eslint-disable-line import/no-default-export
 	constructor(
+		@Inject(DI.meta)
+		private serverSettings: MiMeta,
+
 		private utilityService: UtilityService,
 		private userEntityService: UserEntityService,
 		private noteEntityService: NoteEntityService,
 		private apResolverService: ApResolverService,
 		private apDbResolverService: ApDbResolverService,
 		private apPersonService: ApPersonService,
 		private apNoteService: ApNoteService,
+		private remoteUserResolveService: RemoteUserResolveService,
+		private apiLoggerService: ApiLoggerService,
 	) {
 		super(meta, paramDef, async (ps, me) => {
-			const object = await this.fetchAny(ps.uri, me);
+			let object: SchemaType<typeof meta['res']> | null = null;
+			let acct: misskey.acct.Acct | null = null;
+
+			try {
+				acct = misskey.acct.parseAcctOrUrl(ps.uri);
+			} catch (err) {
+				// nothing to do
+			}
+
+			if (!ps.onlyUriFetch && acct) {
```

Comment:
Should a regular URL fetch be tried first before trying an inferred user account?

It seems this can lead to https://example.com/@alice/post123 being resolved to Alice's profile rather than the actual post.

Reply:
というか、 https://example.com/@alice/post123 はparseAcctOrUrlでエラーにするべきかも